src/impl/transact_log_handler.cpp

```diff
@@ -497,6 +499,13 @@ class TransactLogObserver : public TransactLogValidationMixin, public MarkDirtyM
         return true;
     }
 
+    bool add_row_with_key(size_t row_ndx, size_t prior_num_rows, size_t key_col, int64_t)
+    {
+        insert_empty_rows(row_ndx, 1, prior_num_rows, false);
+        m_active_table->modify(row_ndx, key_col);
```

Comment:
The modification from set_int() on a newly-inserted row is spurious and eventually gets filtered out. SetUnique and SetDefault skip marking the row as modified because they can only ever be called on newly inserted rows, but checking if a row is newly inserted in normal set() ends up being a pessimization.